packages/babel-parser/src/options.ts

```diff
@@ -104,5 +108,24 @@ export function getOptions(opts?: Options | null): OptionsWithDefaults {
   for (const key of Object.keys(defaultOptions) as (keyof Options)[]) {
     options[key] = opts[key] ?? defaultOptions[key];
   }
+
+  if (options.startColumn > 0 && opts.startIndex == null) {
+    if (options.startLine === 1) {
+      options.startIndex = options.startColumn;
+    } else if (process.env.BABEL_8_BREAKING) {
+      throw new Error(
+        "When `startColumn` is used with a `startLine > 1` you must also provide a `startIndex`.",
```

Comment:
When `startLine` is 2 and `startColumn` is 0, we should still throw if `startIndex` is undefined.

Reply:
@JLHwung are you suggesting just changing `options.startColumn > 0` to `options.startColumn >= 0`?

Reply:
👍 That should work for this case.

Reply:
Awesome, I've updated the pr 😄

Reply:
@JLHwung I've refactored the default/error checking code slightly and added another test.

I _think_ it errors in all the cases we want now.